sql/core.go

```diff
@@ -620,6 +627,12 @@ type TableCreator interface {
 	CreateTable(ctx *Context, name string, schema Schema) error
 }
 
+type TemporaryTableCreator interface {
```

Comment:
Interface doc: "TemporaryTableCreator is a database that can create temporary tables that persist only as long as the session"

Also embed the Database interface